File: crud-python/db_connection.py
#!/usr/bin/env python3      
from cassandra.cluster import Cluster
from cassandra.auth import PlainTextAuthProvider

# This is the Zip file you downloaded
SECURE_CONNECT_BUNDLE = '/workspace/workshop-crud-with-python-and-node/crud-python/creds.zip'
# This is the username, recommended value was SUser
USERNAME = "SUser";
# This is the password, recommended value was SPassword1
PASSWORD = "SPassword1";
# This is the keyspace name, recommended value was spacecraft
KEYSPACE = "spacecraft"; 

class Connection:
    def __init__(self):
        self.secure_connect_bundle=SECURE_CONNECT_BUNDLE
        self.path_to_creds=''
        self.cluster = Cluster(
            cloud={
                'secure_connect_bundle': self.secure_connect_bundle
            },
            auth_provider=PlainTextAuthProvider(USERNAME, PASSWORD)
        )
        self.session = self.cluster.connect(KEYSPACE)
    def close(self):
        self.cluster.shutdown()
        self.session.shutdown()

File: crud-nodejs/db_connection.js
const cassandra = require("cassandra-driver");
const TimeUuid = require("cassandra-driver").types.TimeUuid;

// This is the Zip file you downloaded
const SECURE_CONNECT_BUNDLE =
  "/workspace/crud-with-python-and-node/crud-nodejs/creds.zip";

// Update the values with the ones you have
const CLIENT_ID = "kxPbavsfqWoZaBxqHceZxxeq";
const CLIENT_SECRET =
  "tCP7.jnWxa1YHyxsblH4nd2ymsct9UgZj+RxF6d0ruKB1f,z-EdPOL3b2K4w6OQkhdMbkoCMk77D4sNuqXXe4TBRvuLDQS3wkwuTY8uk-5sb+ELC0d8Wws.GCSWPmqe4";
const TOKEN =
  "AstraCS:kxPbavsfqWoZaBxqHceZxxeq:0dfd26f25273e1b6a98318cfc915eb21ad2ccf827518abd379a8f76c4585fe13";

// This is the username, recommended value was SUser
const USERNAME = "SUser";
// This is the password, recommended value was SPassword1
const PASSWORD = "SPassword1";
// This is the keyspace name, recommended value was spacecraft
const KEYSPACE = "spacecraft";

// Init the connection and return the client
function init_connection() {
  var connection = {};
  connection.client = new cassandra.Client({
    cloud: { secureConnectBundle: SECURE_CONNECT_BUNDLE },
    keyspace: KEYSPACE,
    credentials: { username: USERNAME, password: PASSWORD },
    authProvider: new cassandra.auth.PlainTextAuthProvider(
      CLIENT_ID,
      CLIENT_SECRET
    ),
  });
  return connection;
}

connection = init_connection();

module.exports = connection;
